### PR TITLE
Update README: update supported versions + restructure

### PR DESCRIPTION
**Pull request by haroune-mohammedi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,12 +25,23 @@ Abi-wan-kanabi is an UNLICENSE standalone TypeScript parser for Cairo smart cont
...
+### Cairo versions
+
+Abiwan will support multiple Cairo compiler versions, but not in parallel - different package versions will support consecutive Cairo versions.
+
+Currently supported:
+
+| Abi-Wan npm                                                   | Cairo compiler                                                                                                                                               |
+| ------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| [1.0.3](https://www.npmjs.com/package/abi-wan-kanabi/v/1.0.3) | [Cairo v1.0.0](https://github.com/starkware-libs/cairo/releases/tag/v1.0.0) <br> [Cairo v1.1.0](https://github.com/starkware-libs/cairo/releases/tag/v1.1.0) |
+| [2.1.0](https://www.npmjs.com/package/abi-wan-kanabi/v/2.1.0) | [Cairo v2.3.0](https://github.com/starkware-libs/cairo/releases/tag/v2.3.0)                                                                                  |
```
Comment (ivpavici) [inline README.md:33]: what about [v2.4.0](https://github.com/starkware-libs/cairo/releases/tag/v2.4.0) ? I think it should work as well, no ABI changes? If so we can note it here!

File: README.md [hunk 2]
```diff
@@ -42,61 +53,53 @@ Also, it makes use of BigInt, so the tsconfig.json should target at least ES2020
...
-| [1.0.3](https://www.npmjs.com/package/abi-wan-kanabi/v/1.0.3) | [Cairo v1.0.0](https://github.com/starkware-libs/cairo/releases/tag/v1.0.0) <br> [Cairo v1.1.0](https://github.com/starkware-libs/cairo/releases/tag/v1.1.0) |
-| [2.0.0](https://www.npmjs.com/package/abi-wan-kanabi/v/2.0.0) | [Cairo v2.3.0-rc0](https://github.com/starkware-libs/cairo/releases/tag/v2.3.0-rc0)                                                                          |
-
 ### Usage with `starknet.js`
 
-Let's say you want to interact with the [Ekubu: Core](https://starkscan.co/contract/0x00000005dd3d2f4429af886cd1a3b08289dbcea99a294197e9eb43b0e0325b4b) contract using starknet.js, using abiwan you can get the correct types for the contract's functions
+Let's say you want to interact with the [Ekubu: Core](https://starkscan.co/contract/0x00000005dd3d2f4429af886cd1a3b08289dbcea99a294197e9eb43b0e0325b4b) contract using starknet.js, using Abiwan you can get the correct types for the contract's functions
```
Comment (ivpavici) [inline README.md]:
```suggestion
Let's say you want to interact with the [Ekubo: Core](https://starkscan.co/contract/0x00000005dd3d2f4429af886cd1a3b08289dbcea99a294197e9eb43b0e0325b4b) contract using starknet.js, using Abiwan you can get the correct types for the contract's functions
```

**Review by thomas-quadratic — APPROVED** [review]
Great, thanks

File: README.md (hunk 1 above)
Comment (haroune-mohammedi) [reply README.md:33]: Let me check if it works, if so, I'll update it

Comment (haroune-mohammedi) [reply README.md:33]: Even though there is no ABI change, they introduced new types https://github.com/starknet-io/starknet.js/pull/891

**Pull request merged**